azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -123,7 +122,7 @@ func resourceArmSynapseSqlPool() *schema.Resource {
 				ConflictsWith: []string{"restore"},
 				ValidateFunc: validation.Any(
 					validate.SqlPoolID,
-					sqlValidate.SqlDatabaseID,
+					mssqlValidate.MsSqlDatabaseID,
```

Comment:
Same here, is this for all Ms Sql Databases or is it just for synapse sql pool's? If it's just for pool's I would change the name here as well.